src/prepared_discover/prepared_discover_txn.c

```diff
@@ -94,6 +94,58 @@ __prepared_discover_find_or_create_item(WT_SESSION_IMPL *session, uint64_t prepa
     return (0);
 }
 
+/*
+ * __restore_upd_to_ingest_table --
+ *     In disaggregated storage, in follower mode, stable table cannot be modified, therefore a
+ *     prepared update needs to be restored onto ingest table so that the follower node can then
+ *     commit the prepard transaction. This function open the ingest table and insert the update
+ *     restored from disk onto the ingest table. It also sets the session's dhandle to the ingest
+ *     table to track the correct btree when committing/rolling back.
+ */
+static int
+__restore_upd_to_ingest_table(WT_SESSION_IMPL *session, WT_UPDATE *upd, WT_ITEM *key)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_CURSOR *cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_DECL_RET;
+    WT_LAYERED_TABLE_MANAGER *manager;
+    WT_LAYERED_TABLE_MANAGER_ENTRY *entry;
+    uint32_t i, table_count;
+
+    const char *cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_open_cursor), "overwrite", NULL, NULL};
+
+    entry = NULL;
+    conn = S2C(session);
+
+    manager = &conn->layered_table_manager;
+    /* Find the first available layered table entry */
+    table_count = manager->open_layered_table_count;
+    for (i = 0; i < table_count; i++) {
+        /* Find the first non-empty entry */
+        if (manager->entries[i] != NULL) {
+            entry = manager->entries[i];
+            break;
+        }
+    }
+    if (entry == NULL) {
+        ret = WT_NOTFOUND;
+        goto err;
+    }
+    /* Open cursor on the ingest table */
+    WT_ERR(__wt_open_cursor(session, entry->ingest_uri, NULL, cfg, &cursor));
+
+    cbt = (WT_CURSOR_BTREE *)cursor;
+    WT_WITH_PAGE_INDEX(session, ret = __wt_row_search(cbt, key, true, NULL, false, NULL));
+    WT_ERR(ret);
+    WT_ERR(__wt_row_modify(cbt, key, NULL, &upd, WT_UPDATE_INVALID, true, true));
+    ret = __wt_session_get_dhandle(session, entry->ingest_uri, NULL, NULL, 0);
```

Comment:
We get the dhandle and then do nothing?

Reply:
`__wt_session_get_dhandle` sets `session->dhandle` to the ingest_uri.

Later in `__wt_pending_prepared_next_op`, we create a new operation and will assign the ingest table uri as the op's btree
```
    btree = S2BT(session);
    op->btree = btree;
```

it's a bit hard to keep track since these code are far away from each other, but I havent found a better way to refactor this.